packages/graphql-dynamodb-transformer/src/definitions.ts

```diff
@@ -10,28 +11,132 @@ import {
+                pMap.set(def.name.value, def)
+                getNonModelFields(def, ctx, pMap)
+            }
+        }
+    }
+
+    return Array.from(pMap.values())
+}
+
+export function makeNonModelInputObject(
+    obj: ObjectTypeDefinitionNode,
+    nonModelTypes: ObjectTypeDefinitionNode[],
+    ctx: TransformerContext
+): InputObjectTypeDefinitionNode {
+    const name = ModelResourceIDs.NonModelInputObjectName(obj.name.value)
+    const fields: InputValueDefinitionNode[] = obj.fields
+        .filter((field: FieldDefinitionNode) => {
+            const fieldType = ctx.getType(getBaseType(field.type))
+            if (field.name.value === 'id') {
+                return false;
+            }
+            if (
+                isScalar(field.type) ||
+                nonModelTypes.find(e => e.name.value === getBaseType(field.type)) ||
+                (fieldType && fieldType.kind === 'EnumTypeDefinition')
+            ) {
+                return true;
+            }
+            return false;
+        })
+        .map(
+            (field: FieldDefinitionNode) => {
+                const type = nonModelTypes.find(e => e.name.value === getBaseType(field.type)) ?
+                    withNamedNodeNamed(field.type, ModelResourceIDs.NonModelInputObjectName(getBaseType(field.type))) :
+                    field.type
+                return {
+                    kind: Kind.INPUT_VALUE_DEFINITION,
+                    name: field.name,
+                    type: type,
+                    // TODO: Service does not support new style descriptions so wait.
+                    // description: field.description,
+                    directives: []
+                }
+            }
+        )
+    return {
+        kind: 'InputObjectTypeDefinition',
+        // TODO: Service does not support new style descriptions so wait.
+        // description: {
+        //     kind: 'StringValue',
+        //     value: `Input type for ${obj.name.value} mutations`
+        // },
+        name: {
+            kind: 'Name',
+            value: name
+        },
+        fields,
+        directives: []
+    }
+}
+
+export function makeCreateInputObject(
+    obj: ObjectTypeDefinitionNode,
+    nonModelTypes: ObjectTypeDefinitionNode[],
+    ctx: TransformerContext
+): InputObjectTypeDefinitionNode {
     const name = ModelResourceIDs.ModelCreateInputObjectName(obj.name.value)
     const fields: InputValueDefinitionNode[] = obj.fields
-        .filter((field: FieldDefinitionNode) => field.name.value !== 'id' && isScalar(field.type))
+        .filter((field: FieldDefinitionNode) => {
+            const fieldType = ctx.getType(getBaseType(field.type))
+            if (field.name.value === 'id') {
+                return false;
+            }
+            if (
+                isScalar(field.type) ||
+                nonModelTypes.find(e => e.name.value === getBaseType(field.type)) ||
+                (fieldType && fieldType.kind === 'EnumTypeDefinition')
```

Comment:
Use `Kind.ENUM_TYPE_DEFINITION` instead of the hard coded string. You can import `Kind` from `graphql`